BrainCloudClient/Assets/BrainCloud/Client/BrainCloud/BrainCloudLobby.cs

```diff
@@ -550,66 +580,134 @@ private void onRegionForLobbiesSuccess(string in_json, object in_obj)
 
 #if DOT_NET
-        private void PingUpdateSystem(string in_region, string in_target)
+        private void HandleHTTPResponse(string in_region, string in_target)
         {
-            System.Net.NetworkInformation.Ping pinger = new System.Net.NetworkInformation.Ping();
+            if (!in_target.StartsWith("http"))
+            {
+                in_target = (UseHttps ? "https://" : "http://") + in_target;
+            }
+
+            DateTime RoundtripTime = DateTime.UtcNow;
+
+            HttpClient client = new HttpClient();
+            client.Timeout = new TimeSpan(100000000); // 10 seconds
+
+            client.GetAsync(in_target).ContinueWith((Task<HttpResponseMessage> task) =>
+            {
+                if (task.IsCompletedSuccessfully && task.Result is HttpResponseMessage response && response.IsSuccessStatusCode)
+                {
+                    handlePingTimeResponse((long)(DateTime.UtcNow - RoundtripTime).TotalMilliseconds, in_region);
+                }
+                else
+                {
+                    pingNextItemToProcess();
+                }
+
+                client.Dispose();
+            });
+        }
+
+        private void HandlePingReponse(string in_region, string in_target)
+        {
+            Ping pinger = new Ping();
             try
             {
-                pinger.PingCompleted += (o, e) =>
+                pinger.PingCompleted += (o, response) =>
                 {
-                    if (e.Error == null && e.Reply.Status == System.Net.NetworkInformation.IPStatus.Success)
+                    if (response.Error == null && response.Reply.Status == IPStatus.Success)
                     {
-                        handlePingTimeResponse(e.Reply.RoundtripTime, in_region);
+                        handlePingTimeResponse(response.Reply.RoundtripTime, in_region);
+                    }
+                    else
+                    {
+                        pingNextItemToProcess();
                     }
                 };
 
-                pinger.SendAsync(in_target, null);
-            }
-            catch (System.Net.NetworkInformation.PingException)
-            {
-                // Discard PingExceptions and return false;
+                pinger.SendPingAsync(in_target, 10000);
             }
+            catch (Exception) { }
             finally
             {
-                if (pinger != null)
-                {
-                    pinger.Dispose();
-                }
+                pinger?.Dispose();
             }
         }
 #else
+        private IEnumerator HandleHTTPResponse(string in_region, string in_target)
+        {
+            if (!in_target.StartsWith("http"))
+            {
+                in_target = (UseHttps ? "https://" : "http://") + in_target;
+            }
+
+            DateTime RoundtripTime = DateTime.UtcNow;
+#if USE_WEB_REQUEST
+            UnityWebRequest request = UnityWebRequest.Get(in_target);
+            request.timeout = 10;

```

Comment:
Timeouts for the PING/HTTP requests are currently hard-coded to 10 seconds. That is way more than enough time but if we could also add the ability to control this, either by the server or the client themselves can add a setting.